src/main/java/com/clevertap/stormdb/StormDB.java

```diff
@@ -60,6 +61,8 @@
 
     private BitSet dataInWalFile = new BitSet();
 
+    private TIntIntHashMap keyInMemoryBuffer = new TIntIntHashMap();
```

Comment:
Can we use the existing bitset + index strategy to derive whether a key is in the in memory buffer or not?

Reply:
Yes, we can use bitset but there is one problem with bitset and index. Suppose there are only 3 keys in our In-memory buffer 1, 100 and 10000, then TintintHashMap will have (96 bits for keys(4 byte key) and 96 bits for values = 192 bits). But in the case of our bitset, it will be 10048 bits.

Reply:
Well yes, the bitset will be huge. That however, is a different problem to solve (bitset words which are not really required). We should ideally switch to using a more efficient bitset. Perhaps we could create an issue for that and address that separately instead of mixing these two features?

Reply:
Sure but for now I think TintintHashMap is the best choice until we introduce some kind of sparseBitset

Reply:
I don't think so - it introduces two sources of truth in the code itself. The combination of (index, bitset) being the first source of truth, and then the new map which you've introduced.

The problem you mentioned (about wasted words in the bitset) exists today as well. I think we should solve it differently, as even without your new map, the existing bitset would've been [expanded here](https://github.com/CleverTap/stormdb/pull/22/files/6028bfa42b5e9bcc4e05db385eec9edc8a6c9128?diff=split&w=1#diff-f6e18c7df39ad0e0aa3bdbb6b0b2e01bL524).

We ideally do not need any additional data structure to support the addition of an in-place update, right? Or is there something else here which I've overlooked?

Reply:
I forgot to consider that dataInWalFile index because according to its name it means data in the current wal file not in memory buffer. I think I can use this in combination with `index` map